core/src/main/java/com/linecorp/armeria/unsafe/ByteBufHttpData.java

```diff
@@ -35,6 +37,69 @@
 /**
  * A {@link HttpData} that is backed by a {@link ByteBuf} for optimizing certain internal use cases. Not for
  * general use.
+ *
+ * <p>The buffer backing a {@link ByteBufHttpData} is a pooled buffer - this means that it does not use normal
+ * Java garbage collection, and instead uses manual memory management using a reference count, similar to
+ * some constructs in languages like C++. Manual memory management is more fragile and not idiomatic in Java -
+ * you should only use this class in performance-sensitive situations and after being ready to deal with these
+ * very hard-to-debug issues.
+ *
+ * <p>You may interact with {@link ByteBufHttpData} when using methods that return pooled objects, such as
+ * {@link com.linecorp.armeria.common.HttpResponse#aggregateWithPooledObjects(ByteBufAllocator)}. If you don't
+ * use such methods, you will never see a {@link ByteBufHttpData} and don't need to read further.
+ *
+ * <p>Any time you receive a {@link ByteBufHttpData} it will have a single reference that must be released -
+ * failure to release the reference will result in a memory leak and poor performance. You will want to make
+ * sure to do this by calling {@link HttpData#close()}, usually in a try-with-resources structure.
+ *
+ * <p>For example, <pre>{code
+ *
+ * HttpResponse response = client.get("/");
+ * response.aggregateWithPooledObjects(ctx.alloc(), ctx.executor())
+ *     .thenApply(aggResp -> {
+ *        try (HttpData content = aggResp.content()) {
```

Comment:
add comment above try, as there is a lot of code to tell what line is important.

```
// HERE: try with resources on the content avoids reliance on downstream calls to do the same.
```